src/site/content/en/blog/preload-optional-fonts/index.md

```diff
@@ -0,0 +1,110 @@
+
+Layout shifting, or re-layout, occurs when a resource on a web page changes dynamically, resulting in
+a "shift" of content. Fetching and rendering web fonts can directly cause layout shifts in one of
+two ways:
+
+-  A fallback font is swapped with a new font ("flash of unstyled text")
+-  "Invisible" text is shown until a new font is rendered into the page ("flash of invisible text")
+
+The CSS [`font-display`](https://font-display.glitch.me/) property provides a way to modify
+rendering behavior of custom fonts through a range of different supported values (`auto`, `block`,
+`swap`, `fallback`, and `optional`). Choosing which value to use depends on the preferred behavior
+for asynchronously loaded fonts. However, every one of these supported values can trigger re-layout
+in one of the two ways listed above, until now!
+
+{% Aside %}
+  The [Cumulative Layout Shift](https://web.dev/cls/) metric makes it possible to measure the layout
+  instability on a web page.
+{% endAside %}
+
+## Optional fonts
+
+The `font-display` property uses a timeline of three periods to handle fonts that need to be
+downloaded before they can be rendered:
+
+-  **Block:** Render "invisible" text, but switch to the web font as soon as it finishes loading.
+-  **Swap:** Render text using a fallback system font, but switch to the web font as soon as it
+   finishes loading.
+-  **Fail:** Render text using a fallback system font.
+
+Previously, fonts designated with `font-display: optional` had a 100ms block period and no swap
+period. This means that "invisible" text is displayed very briefly before switching to a fallback
+font. If the font is not downloaded within 100ms, then the fallback font is used and no swapping
+occurs.
+
+<figure class="w-figure">
+  <img src="./previous-behavior-fail.png" alt="Diagram showing previous optional font behavior when font fails to load">
+  <figcaption class="w-figcaption">Previous <code>font-display: optional</code> behavior in Chrome when font is downloaded <b>after</b> the 100ms block period</figcaption>
+</figure>
+
+However, in the case that the font is downloaded before the 100ms block period completes, the custom
+font is rendered and used on the page.
+
+<figure class="w-figure">
+  <img src="./previous-behavior-pass.png" alt="Diagram showing previous optional font behavior when font loads in time">
+  <figcaption class="w-figcaption">Previous <code>font-display: optional</code> behavior in Chrome when font is downloaded <b>before</b> the 100ms block period</figcaption>
+</figure>
+
+Chrome re-renders the page **twice** in both instances, regardless of whether the fallback font
+is used or if the custom font finishes loading in time. This causes a slight flicker of invisible
+text and, in cases when a new font is rendered, layout jank that moves some of the page's content.
+This occurs easily if the font is stored in the browser's disk cache and can load well before the
+block period is complete (approximately 10ms).
```

Comment:
```suggestion
This occurs even if the font is stored in the browser's disk cache and can load well before the
block period is complete (approximately 100ms).
```
I think this should be "occurs even" because it's pointing out a problem in the previous implementation? Also, elsewhere in the article it refers to 100ms as the block period?

Reply:
Ah good catch. Fixed to "occurs even" 👍 

The 10ms is in reference to how long it can take to load from the cache. But it is kind of confusing to mention it right after I re-iterate the block period, so I just removed it :)